app/services/oidc_jwt_validate.rb

```diff
@@ -0,0 +1,48 @@
+class OidcJwtValidate
+  attr_reader :decoded_token
+  delegate :logger, to: :Rails
+
+  def initialize(jwt_token)
+    @jwt_token = jwt_token
+  end
+
+  def decoded_payload
+    # OpenSSL#set_key method does not support ruby version < 2.4.0, apparently the JWT gem uses
+    # OpenSSL#set_key method for all ruby version. We must remove this condition once new version
+    # of the JWT(2.2.2) is released.
+    unless OpenSSL::PKey::RSA.new.respond_to?(:set_key)
+      Foreman::Logging.logger('app').error "SSO feature is not available for Ruby < 2.4.0"
+      return nil
+    end
+    return nil if Setting['oidc_algorithm'].nil?
+    JWT.decode(@jwt_token, nil, true,
+      { iss: Setting['oidc_issuer'],
+        verify_iss: true,
+        algorithms: [Setting['oidc_algorithm']],
```

Comment:
What happend to `aud` and `verify_aud`?

Reply:
Sorry, this happened by mistake, I will add this with other suggested changes. Thanks for noticing.